fearless/Modules/Pincode/CheckPincode/CheckPincodePresenter.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+class CheckPincodePresenter: PinSetupPresenterProtocol {
+    weak var view: PinSetupViewProtocol?
+    let interactor: LocalAuthInteractorInputProtocol
+    let moduleOutput: CheckPincodeModuleOutput
+
+    init(
+        interactor: LocalAuthInteractorInputProtocol,
+        moduleOutput: CheckPincodeModuleOutput
+    ) {
+        self.interactor = interactor
+        self.moduleOutput = moduleOutput
+    }
+
+    func start() {
+        view?.didChangeAccessoryState(
+            enabled: interactor.allowManualBiometryAuth,
+            availableBiometryType: interactor.availableBiometryType
+        )
+        interactor.startAuth()
+    }
+
+    func cancel() {}
+
+    func activateBiometricAuth() {
+        interactor.startAuth()
+    }
+
+    func submit(pin: String) {
+        interactor.process(pin: pin)
+    }
+}
+
+extension CheckPincodePresenter: LocalAuthInteractorOutputProtocol {
+    func didEnterWrongPincode() {
+        DispatchQueue.main.async { [weak self] in
+            self?.view?.didReceiveWrongPincode()
+        }
+    }
+
+    func didChangeState(from _: LocalAuthInteractor.LocalAuthState) {}
+
+    func didCompleteAuth() {
+        DispatchQueue.main.async { [weak self] in
+            self?.moduleOutput.didCheck()
+        }
+    }
+
+    func didUnexpectedFail() {}
```

Comment:
shouldn't we present error here?

Reply:
We show error in didEnterWrongPincode() method. "didUnexpectedFail()" method was used only for case then user totally forgot his pin and want to sign up again.